rasa/core/policies/ted_policy.py

```diff
@@ -927,15 +925,11 @@ def _check_data(self) -> None:
 
     def _prepare_layers(self) -> None:
```

Comment:
I think we aren't adding the loss term for masked language modelling if it's switched on in the config.

Reply:
The masking for it is done inside `RasaSequenceLayer` I believe but the loss term is not added. It's done by `_prepare_mask_lm_loss` in DIET. I guess it's not done here anywhere or am I missing something?

Reply:
we decided not to add this term to ted, since it showed that it doesn't seem to help. The fact that masking is performed in ted is more of an artifact of shared methods, but also can be used as a regularizer

Reply:
Not sure, if for TED it would be a good regularizer or add noise since the sequence of input states are actually important for the action prediction. For intent classification, all input tokens do not matter as much and hence it's okay to mask some.
I think it's a separate discussion probably. @samsucik Could you please add a comment [here](https://github.com/RasaHQ/rasa/blob/05ceab64e059c2fc0cd1cec288dcf9bbcc7532c1/rasa/core/policies/ted_policy.py#L258) and where we are preparing the masked input, that for TED it will only act as an input dropout since there is no loss term added.

Reply:
@dakshvar22 sure!

Reply:
> Not sure, if for TED it would be a good regularizer or add noise since the sequence of input states are actually important for the action prediction.
Importantly, MLM is only done on the `TEXT` attribute (this is hard-coded in `rasa_layers`).